### ISSUE TITLE
MBR on EXT4: Blank screen after loading message?

### ISSUE THREAD

**Issue by SomethingGeneric** [Limine-Bootloader/Limine#141]
**Limine version used**
2.68.2 (binary branch)

**Expected behavior**
BusyBox init to load and thus bring up the system

**Actual (broken) behaviour**
Blank screen? I think the system is up since there's CPU usage in the VM

**To Reproduce**
Try booting `./build/ikeda` after running `./build.sh qemu` from https://git.tar.black/ikeda/Ikeda/-/tree/limine (ensure `limine` branch :)  )
Expected behavior can be seen doing the same with the `main` branch (using GRUB2)

**Stacktrace**
N/A

**Additional context**
I saw another closed issue about VM memory, so I tried using virt-manager instead of just pure qemu

`/boot/limine.cfg` contents:
```
PROTOCOL=linux
KERNEL_PATH=boot:///boot/bzImage
KERNEL_CMDLINE=quiet
```
(same behavior w/o `quiet` param.)

I'm happy to share the source disk image(s) in question, just not sure how to do so well since they're 2.9GB

**Comment by SomethingGeneric** [comment]
(edited to fix formatting goof)

**Comment by leap0x7b** [comment]
Looks like it also happened when loading linux kernels in other file systems like ISO 9660

**Comment by leap0x7b** [comment]
Can you try to get the output of `-serial stdio` and `-debugcon stdio` in QEMU?

**Comment by Mintsuki** [comment]
I cannot reproduce this by simply attempting to load a Linux kernel from an ext4 partition.
Please send over the image used to reproduce this by whatever means you want (may I suggest maybe Google Drive?)

**Comment by Mintsuki** [comment]
> Looks like it also happened when loading linux kernels in other file systems like ISO 9660

I have no idea what bug is being referenced here, but if you have some reproducible example, please share it, also keep it on topic.

> Can you try to get the output of `-serial stdio` and `-debugcon stdio` in QEMU?

That will not work as the prebuilt image is not compiled with E9 output enabled.

**Comment by SomethingGeneric** [comment]
@mintsuki here's `ikeda-limine` as a compressed xz of the raw image https://drive.google.com/file/d/1JcntEsKLQUERKqbj6xzno5LLmkoRWWAf/view?usp=sharing

here's `ikeda-grub` as a compressed xz of the image
https://drive.google.com/file/d/1oPehgNVody1OKp1pfhwcs9F8BydlU4D_/view?usp=sharing

**Comment by Mintsuki** [comment]
Your kernel doesn't appear to support a framebuffer console which is what Limine defaults to. As per config documentation, adding `TEXTMODE=yes` reveals the kernel's output, which then reveals a panic due to the root not having been specified.

Specifying the root in the kernel's command line equalises the outputs of GRUB and Limine.

**Comment by SomethingGeneric** [comment]
ah, ok. I'm new to configuring kernels so I guess I missed that. Thanks for the help!

**Comment by Mintsuki** [comment]
You're welcome!

**Comment by leap0x7b** [comment]
> I have no idea what bug is being referenced here, but if you have some reproducible example, please share it, also keep it on topic.

https://www.dropbox.com/s/hflmfvjbp8610ql/test.iso?dl=0

**Comment by leap0x7b** [comment]
In UEFI, the kernel boots fine (and also boots to fbcon), meanwhile in BIOS, it's just stuck at loading kernel

**Comment by Mintsuki** [comment]
> In UEFI, the kernel boots fine (and also boots to fbcon), meanwhile in BIOS, it's just stuck at loading kernel

This wasn't a technical bug per se but just an issue where if an initramfs was gz compressed, it would not show the loading message while decompressing it (fixed in 2be3776).

Now, the image works for me, but you have to keep in mind that gz decompression with tinf is slow, and especially so if not using hardware virtualisation, so it wasn't hanging and there was no issue with ISO or anything else, it was just decompressing without showing a loading message and taking a long time because you were not using hardware virtualisation.

I recommend not using gz compression unless you really need to save space on disk, as the speed tradeoff is usually not worth it.
